### ISSUE TITLE
Can't link to XTouch

### ISSUE THREAD

**Issue by dmontoya3** [bramdebouvere/fltouch#15]
I downloaded the zip, copied everything to D:\Libraries\Documents\Image-Line\Data\FL Studio\Settings\Hardware\fltouch
(I moved my libraries to a separate HDD since I have Windows on a SSD but with less capacity)

When I open FL studio it shows 4 new device types 
FLTouch X-touch (user)
FLTouch X-touch (factory)
FLTouch X-touch Extender (user)
FLTouch X-touch Extender (factory)

![xtouch 01](https://user-images.githubusercontent.com/41489098/178070549-10ae8b1a-0109-4f56-917a-ec1b874a3b12.png)

I'm trying both FLTouch X-touch (user) and (factory), set the MIDI channel to 102 and tried activating/deactivating the "Send master sync" button

![image](https://user-images.githubusercontent.com/41489098/178070733-5b9e834f-af1c-4195-a1c9-50018f974db0.png)

After that, the XTouch doesn't respond. If I move a fader, it goes back to its original position and no button seems to work at all

If I go back to Mackie Control Universal, the XTouch works as expected

XTouch is in Mode = MC and IFC = USB
I'm using Windows 11

Not sure if the fact that FL Studio is installed in C: and all the user data is in D: is interfering

**Comment by bramdebouvere** [comment]
Hey there, sorry for the late response. Didn't get any notification.

I'm pretty sure putting it on another drive is no issue, since I'm also loading it from "E:\Documents\Image-Line\Data\FL Studio\Settings\Hardware\fltouch" myself. "Factory" is usually reserved for scripts that are included with FL studio.

"Send master sync" does not need to be enabled, as the script sends everything that is required.

That being said, I find it very weird that you see 2 extra options with "factory", as I only see the "FLtouch X-Touch (user)" and "FLtouch X-Touch Extender (user)" myself here.

Can you please check:
- What's the output you get when you go to "View -> Script Output" and select the "X-Touch" tab? Do you see any errors?
- Are you running the latest version of FL Studio? (I notice some settings are missing from your screenshot that I have in my version). It's possible that you are running an older version and that the script is using some functionality that is only available in a later version.
- Have you updated the X-touch firmware to the latest version?

**Comment by dmontoya3** [comment]
FACTORY VS USER
I now know why I have two versions of each (factory and user). This is happening because I tried installing the script in several folders, one in 
C:\Program Files\Image-Line\FL Studio 20\System\Hardware specific\fltouch 
and one in 
D:\Libraries\Documents\Image-Line\FL Studio\Settings\Hardware\fltouch

FL studio is considering the ones in C as factory and the ones in D as user but it doesn't interfere. I tried removing one, trying one or the other and the result is the same.

FIRMWARE
Regarding the XTouch firmware I did updated it because apparently it wasn't compatible with AMD CPUs so at first I wasn't able to use it at all. Need to double check if there's been a newer release since I did it a couple of months ago. 

VERSION
I'm running FL Studio 20.6.2 [Build 1549]. 

SCRIPT OUTPUT
Don't have "Script output" menu but "Python output". After selecting the FL Touch X-touch I get this in the output:

![image](https://user-images.githubusercontent.com/41489098/183678619-17f5ffa1-4f3e-407a-8922-16300acc3f02.png)

If I click "Edit Script" it opens a notepad with it 
![image](https://user-images.githubusercontent.com/41489098/183678811-f4a6947b-5671-4e9d-ac2f-7b51386b6ee9.png)

These are the files I have on the folder
![image](https://user-images.githubusercontent.com/41489098/183681459-ead22f56-e9f9-4030-8d9c-0220116b8e57.png)

Thank you in advance.
